### PR TITLE
Feature model pipeline

### PR DESCRIPTION
**Pull request by williamteles**
Creating the model as a pipeline with the preprocessing step and model. This pull request is related with Issue #3

### CODE REVIEW

**Comment by Gil32610** [comment]
Hey @williamteles ! The pipeline implementation is really good! :)

During the pipeline testing I came across a parameter named `validation_data` on the [`mlex/mlex/models`](https://github.com/IoTDataAtelier/mlex/tree/feature/model_pipeline/mlex/models) top modules `rnn.py` `lstm.py` and `gru.py` . The default value is None but there is a step on the fit method accessing the parameters validation_data as a list. What exactly is expected as argument to it?

**Comment by williamteles** [comment]
@Gil32610 this column is referred to the context we want to use when creating the sequences. On [`examples/example_pipeline.py`](https://github.com/IoTDataAtelier/mlex/blob/feature/model_pipeline/examples/example_pipeline.py#L25) line 25 you can see an example on how this column is created.

**Comment by williamteles** [comment]
@Gil32610, I made the update to address this issue. The predict was directly calling the predic_proba; I modified it to call score_samples before.

File: mlex/features/sequences.py
Comment (Gil32610) [inline mlex/features/sequences.py:1]: `group_index` and `column_to_stratify` are interchangeable and passed during model fit to ensure valid sequences are built within the same column entry value. When not provided, sequences are generated only within sequence length.

File: mlex/models/base_components/gru_base_model.py
Comment (Gil32610) [inline mlex/models/base_components/gru_base_model.py:1]: `fit()` sets random seeds to ensure different model outputs and calls `__fit_core()` to start the training epochs.   

All base models `predict` methods call `score_samples` to convert the X input into a DataLoader and make the actual sequence predictions.

File: mlex/models/gru.py
Comment (Gil32610) [inline mlex/models/gru.py:1]: `get_y_true_sequences` and `create_test_loader` are auxiliary methods to the models during test. Given that a sequence composition could potentially reduce the transactions quantity (both by transactions not reaching seq_len or reaching adjacent group), these methods return the ground truth values for the test set with their respective sequences.

**Review by Gil32610 — APPROVED** [review]
Pipeline implementation is working! Great job! :+1: :+1: :+1:

File: mlex/models/base_components/gru_base_model.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        output = self.sigmoid(linear_out)
+
+        return output
+
+    @property
+    def name(self):
+        return "GRUBaseModel"
+
+    def fit(self, X, y):
+        torch.cuda.manual_seed(self.random_seed)
```
Comment (diegompin) [inline mlex/models/base_components/gru_base_model.py:94]: Todas essas random são necessárias ?
